src/gpio.rs

```diff
@@ -1026,14 +1016,11 @@ gpio!([
                 AF4: (into_af4, af4,),
                 AF15: (into_af15, af15,),
             ], [
-                AF2: (into_af2, af2, ["stm32f302", "stm32f303xb", "stm32f303xc", "stm32f303xd", "stm32f303xe", "stm32f358", "stm32f378", "stm32f398",],),
-                AF5: (into_af5, af5, ["stm32f378",],),
-                AF6: (into_af6, af6, ["stm32f378",],),
-                AF7: (into_af7, af7, ["stm32f301", "stm32f303xd", "stm32f303xe", "stm32f318", "stm32f334", "stm32f328", "stm32f378", "stm32f398",],),
-                AF8: (into_af8, af8, ["stm32f302", "stm32f303xb", "stm32f303xc", "stm32f303xd", "stm32f303xe",  "stm32f358", "stm32f378", "stm32f398",],),
-                AF9: (into_af9, af9, ["stm32f302", "stm32f303", "stm32f334", "stm32f328", "stm32f358", "stm32f378", "stm32f398",],),
+                AF2: (into_af2, af2, ["stm32f302", "stm32f303xb", "stm32f303xc", "stm32f303xd", "stm32f303xe", "stm32f358", "stm32f398",],),
+                AF7: (into_af7, af7, ["stm32f301", "stm32f303xd", "stm32f303xe", "stm32f318", "stm32f334", "stm32f328", "stm32f398",],),
+                AF8: (into_af8, af8, ["stm32f302", "stm32f303xb", "stm32f303xc", "stm32f303xd", "stm32f303xe",  "stm32f358", "stm32f398",],),
+                AF9: (into_af9, af9, ["stm32f302", "stm32f303", "stm32f334", "stm32f328", "stm32f358", "stm32f398",],),
                 AF10: (into_af10, af10, ["stm32f303xb", "stm32f303xc", "stm32f303xd", "stm32f303xe", "stm32f358", "stm32f398",],),
-                AF11: (into_af11, af11, ["stm32f378",],),
```

Comment:
I'm not sure about this change. I guess you where removing all `stm32f378` instances, when `stm32f373` instances are missing and `stm32f318` instances, when `stm32f301` instances are missing, am I correct? 

But this case is a bit strange. I became suspicious, that only the `stm32f378` instance was there in the first place. 
This logic was added in https://github.com/stm32-rs/stm32f3xx-hal/pull/21 and looking at it, I'm not sure how this line came to be. 

I checked it against the [datasheet](https://www.st.com/resource/en/datasheet/stm32f373cc.pdf) available on [st.com](https://www.st.com/en/microcontrollers-microprocessors/stm32f373.html#resource) and for pin `PB8` exists an alternate function `AF11` / `TIM19_CH3` (as we can see on page 43). But in this case, `stm32f373` should also be in the list. 

I'm not sure now, how to go about this, because this list probably contains more of these errors (if these are errors and I am reading it right). 

IMO we can leave it as is for this PR and open another PR, which rechecks all these functions against the datasheet again :)

Reply:
> I'm not sure about this change. I guess you where removing all `stm32f378` instances, when `stm32f373` instances are missing and `stm32f318` instances, when `stm32f301` instances are missing, am I correct?

That's correct, but I'm also adding new instances of `stm32f378` and `stm32f318` which should make up for the ones removed. For example, the instance you're commenting on, in GPIOB, is added again in line 1097: There is a second GPIOB definition, which was for `stm32f373` only before. Now `stm32f373` is added there too.

So in general, no AF definition should be lost, they are just (slightly) more compact now.